workflow-bot-app/src/main/java/com/symphony/bdk/workflow/management/WorkflowAbstractAction.java

```diff
@@ -0,0 +1,61 @@
+package com.symphony.bdk.workflow.management;
+
+import com.symphony.bdk.workflow.configuration.WorkflowDeployer;
+import com.symphony.bdk.workflow.exception.DuplicateException;
+import com.symphony.bdk.workflow.exception.NotFoundException;
+import com.symphony.bdk.workflow.swadl.SwadlParser;
+import com.symphony.bdk.workflow.swadl.v1.Workflow;
+
+import lombok.RequiredArgsConstructor;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+
+@RequiredArgsConstructor
+public abstract class WorkflowAbstractAction {
+
+  private final WorkflowDeployer deployer;
+
+  protected Workflow convertToWorkflow(String content) {
+    try {
+      return SwadlParser.fromYaml(content);
+    } catch (Exception e) {
+      throw new IllegalArgumentException("SWADL content is not valid");
+    }
+  }
+
+  protected void validateFilePath(String id, String path) {
+    if (deployer.workflowSwadlPath(id) != null && deployer.workflowSwadlPath(id).endsWith(path)) {
```

Comment:
isn't `deployer.workflowSwadlPath(id) != null` enough? if a workflow is already deployed with "id",  we should throw DuplicateException.

Reply:
it s not enough, this check is to avoid overwriting a file having the same file name but different workflow id (e.g. manually dropped in the workflow directory), and the file's name happens to be the same with the received workflow id. The case would be very rare, but could happen.

Reply:
i changed this check, the way was done before was indeed not correct.